### DISCUSSION TITLE
Swarm Lighthouse Deck Calibration

### DISCUSSION THREAD

**Discussion by Jackson214** [bitcraze/discussions#378] [Q&A] [answered]
Is there a way to calibrate a swarm of Crazyflie 2.1's with the Lighthouse deck at the same time? Calibrating them individually with the cfclient takes a long time with a large swarm. By "calibration" I specifically mean the Lighthouse geometry calibration you do on the cfclient. Thanks!

**Comment by krichardsson** [comment, marked as answer]
Hi, 
yes there is! It is actually a bad idea to estimate the geometry for each Crazyflie as they will have a slightly different idea of the global coordinate system. The proper way of doing it is to:
1. Estimate the geometry for one Crazyflie
2. In the lighthouse tab in the client, click the "save system config" button. This will save the geometry and calibration data to file.
3. Connect the second crazyflie to the client and click the "Load system config" button and pick the file that was saved in step 2. Now this Crazyflie will have the same geometry/calib data as the first one
4. Repeat 3. for the rest of the Crazyflies in the swarm 

If you want to automate 3. you can take a look at the demo we did for IROS 2022, [there is a script that uploads the geometry to a swarm](https://github.com/bitcraze/crazyflie-firmware-experimental/blob/iros-2022/examples/demos/decentralized_swarm/write-geo-all.py) that you can use for inspiration.

**Reply by Jackson214** [reply]
Thanks! That script is exactly what I was looking for! I'll be testing it next time I'm in the lab.

**Reply by Jackson214** [reply]
@krichardsson In the IROS 2022 swarm lighthouse updater script, it doesn't completely work for me. After using "write-geo-all.py", I then read the lighthouse calibration data through the non-experimental "read_lighthouse_mem.py" script in the crazyflie-lib-python/examples/lighthouse folder which shows the data was successful in uploading to all crazyflies. 

However, the geometry for base station 1 always has its "valid" flag set to False. This has caused the swarm to drift as they are not using base station 1 for positioning due to the false valid flag. Do you know how to fix this?

If I manually upload the data to each drone using the cfclient and the previously exported calibration data, the valid flag is True and works just fine. I just want to get it working with the automated script for the swarm (write-geo-all.py)
